mapie/time_series_regression.py

```diff
@@ -0,0 +1,312 @@
+        -------
+        NDArray
+            Array of betas minimizing the differences
+            ``(1-alpa+beta)-quantile - beta-quantile``.
+
+        Raises
+        ------
+        ValueError
+            If lower and upper bounds arrays don't have the same shape.
+        """
+        if lower_bounds.shape != upper_bounds.shape:
+            raise ValueError(
+                "Lower and upper bounds arrays should have the same shape."
+            )
+        alpha = cast(NDArray, alpha)
+        betas_0 = np.full(
+            shape=(len(alpha), len(lower_bounds)),
+            fill_value=np.nan,
+            dtype=float,
+        )
+
+        for ind_alpha, _alpha in enumerate(alpha):
+            betas = np.linspace(
+                _alpha / (len(lower_bounds) + 1),
+                _alpha,
+                num=len(lower_bounds),
+                endpoint=True,
+            )
+            one_alpha_beta = np_nanquantile(
+                upper_bounds,
+                1 - _alpha + betas,
+                axis=1,
+                method="higher",
+            )  # type: ignore
+            beta = np_nanquantile(
+                lower_bounds,
+                betas,
+                axis=1,
+                method="lower",
+            )  # type: ignore
+            betas_0[ind_alpha, :] = betas[
+                np.argmin(one_alpha_beta - beta, axis=0)
+            ]
+
+        return betas_0
+
+    def _pred_multi(self, X: ArrayLike) -> NDArray:
+        """
+        Return a prediction per train sample for each test sample, by
+        aggregation with matrix  ``k_``.
+
+        Parameters
+        ----------
+            X: NDArray of shape (n_samples_test, n_features)
+                Input data
+
+        Returns
+        -------
+            NDArray of shape (n_samples_test, n_samples_train)
+        """
+        y_pred_multi = np.column_stack(
+            [e.predict(X) for e in self.estimators_]
+        )
+        # At this point, y_pred_multi is of shape
+        # (n_samples_test, n_estimators_). The method
+        # ``aggregate_with_mask`` fits it to the right size
+        # thanks to the shape of k_.
+
+        y_pred_multi = self.aggregate_with_mask(y_pred_multi, self.k_)
+        return y_pred_multi
+
+    def predict(
+        self,
+        X: ArrayLike,
+        ensemble: bool = False,
+        alpha: Optional[Union[float, Iterable[float]]] = None,
+        beta_optimize: bool = True,
+    ) -> Union[NDArray, Tuple[NDArray, NDArray]]:
+        """
+        Correspond to 'Conformal prediction for dynamic time-series'.
+
+        Parameters
+        ----------
+        beta_optimize: bool
+            Whether to optimize the PIs' width or not.
+
+        """
+
+        # Checks
+        check_is_fitted(self, self.fit_attributes)
+        self._check_ensemble(ensemble)
+        alpha = cast(Optional[NDArray], check_alpha(alpha))
+        X = check_array(X, force_all_finite=False, dtype=["float64", "object"])
```

Comment:
This line has been removed in a PR a couple of months ago in `MapieRegressor` in order to avoid the automatic conversion of non-numpy arrays to numpy arrays so pipelines using the columns of dataframes can be used.

Reply:
Removed from MapieTimeSeriesRegressor too